src/compiler/compiler.lisp

```diff
@@ -1693,26 +1705,14 @@
         (convert-toplevel sexp multiple-value-p return-p))))
   ;; Process as toplevel
   (let ((*convert-level* -1))
-    (cond
-      ;; Non-empty toplevel progn
-      ((and (consp sexp)
-            (eq (car sexp) 'progn)
-            (cdr sexp))
-       `(progn
```

Comment:
Good cleanup. I guess this is is old from before we even had `define-compilation`!